packages/devextreme/testing/tests/DevExpress.ui.widgets/chatParts/messageList.tests.js

```diff
@@ -0,0 +1,297 @@
+        });
+
+        QUnit.test('scrollabel should be rendered inside root element', function(assert) {
+            assert.ok(Scrollable.getInstance(this.$element.children().first()) instanceof Scrollable);
+        });
+
+        QUnit.test('new message group component should be rendered only when a message with a different user ID is encountered', function(assert) {
+            this.reinit({
+                items: [
+                    { author: { id: 'UserID' } },
+                    { author: { id: 'UserID_1' } },
+                    { author: { id: 'UserID' } }
+                ]
+            });
+
+            const $messageGroups = this.$element.find(`.${CHAT_MESSAGE_GROUP_CLASS}`);
+
+            assert.strictEqual($messageGroups.length, 3);
+        });
+
+        QUnit.test('new message group component should not be rendered when a message with a different user ID is encountered', function(assert) {
+            this.reinit({
+                items: [
+                    { author: { id: 'UserID' } },
+                    { author: { id: 'UserID' } },
+                    { author: { id: 'UserID' } }
+                ]
+            });
+
+            const $messageGroups = this.$element.find(`.${CHAT_MESSAGE_GROUP_CLASS}`);
+
+            assert.strictEqual($messageGroups.length, 1);
+        });
+
+        QUnit.test('essage group component should not be rendered if items is empty', function(assert) {
+            const $messageGroups = this.$element.find(`.${CHAT_MESSAGE_GROUP_CLASS}`);
+
+            assert.strictEqual($messageGroups.length, 0);
+        });
+    });
+
+    QUnit.module('Options', function() {
+        QUnit.test('Message list should render 1 new message if items has been changed by it', function(assert) {
+            const { items } = this.instance.option();
+            const newMessage = {
+                timestamp: NOW,
+                author: userFirst,
+                text: 'NEW MESSAGE',
+            };
+
+            this.instance.option({ items: [...items, newMessage] });
+
+            const $bubbles = this.$element.find(`.${CHAT_MESSAGE_BUBBLE_CLASS}`);
+
+            assert.strictEqual($bubbles.length, items.length + 1);
+        });
+
+        QUnit.test('should run invalidate after changing user in runtime', function(assert) {
+            const invalidateStub = sinon.stub(this.instance, '_invalidate');
+
+            this.instance.option({ currentUserId: 'newUserID' });
+
+            assert.strictEqual(invalidateStub.callCount, 1);
+        });
+
+        QUnit.test('should run invalidate after changing items in runtime', function(assert) {
+            const invalidateStub = sinon.stub(this.instance, '_invalidate');
+
+            this.instance.option({ items: [] });
+
+            assert.strictEqual(invalidateStub.callCount, 1);
+        });
+
+        QUnit.test('Message list should run invalidate if new value is empty', function(assert) {
```

Comment:
Let's create a module like "items option change performance" and move invalidateSpy to beforeEach? It'll be easier to groom these tests in the future

Reply:
https://github.com/DevExpress/DevExtreme/pull/27942/commits/d0167a062af62e594c1ca64a74ab97e9d7bdb528